src/view_pos_transform.hpp

```diff
@@ -34,13 +34,13 @@ class view_pos_transform : public ranges::view_base
 
 private:
     // The data members of view_pos_transform.
-    urng_t                              urange;
-    std::shared_ptr<pos_transform_t>    pos_transform = nullptr;
-    std::shared_ptr<size_transform_t>   size_transform = nullptr;
+    urng_t                                        urange;
+    ::ranges::semiregular_box_t<pos_transform_t>  pos_transform;
+    ::ranges::semiregular_box_t<size_transform_t> size_transform;
 
     // Associated types iterator.
-    using reference         = decltype((*pos_transform)(urange, 0));
-    using const_reference   = decltype((*pos_transform)(std::as_const(urange), 0));
+    using reference         = decltype(pos_transform(urange, 0));
```

Comment:
I thought it needs to stay :wink:

Reply:
Yeah. So. semiregular_box is an addition to C++20. But only *exposion-only* which means that it's behaviour is specified and other components rely on it, but it's not actually there. The way it is specified is that it requires `operator*`, but the version that is in the range-v3 library does not require it 🤷🏻